### PR TITLE
style(page): Refactored code to remove redundancy

### PR DESCRIPTION
**Pull request by yashLadha**
There were multiple if block in the logic that can be ironed
out into a single statement, which is much more robust and readable.

### CODE REVIEW

**Comment by yashLadha** [comment]
There is some flakiness around `accessibility.spec` which is probably targeted in this pr: https://github.com/GoogleChrome/puppeteer/pull/5116

**Comment by yashLadha** [comment]
@aslushnikov  Can you please review the PR?

File: test/accessibility.spec.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ module.exports.addTests = function({testRunner, expect, FFOX}) {
         </select>
       </body>`);
 
+      await page.focus('[placeholder="Empty input"]');
```
Comment (mathiasbynens) [inline test/accessibility.spec.js:45]: Please remove these unrelated changes from this patch

File: lib/Page.js [hunk 2]
```diff
@@ -892,10 +892,8 @@ class Page extends EventEmitter {
       screenshotType = options.type;
     } else if (options.path) {
       const mimeType = mime.getType(options.path);
-      if (mimeType === 'image/png')
-        screenshotType = 'png';
-      else if (mimeType === 'image/jpeg')
-        screenshotType = 'jpeg';
+      const mimeMatch = mimeType.match(/^image\/(png|jpeg)$/);
+      screenshotType = Array.isArray(mimeMatch) ? mimeMatch[1] || null : null;
```
Comment (mathiasbynens) [inline lib/Page.js:896]: This is less readable than the original code.
